src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -2738,8 +2738,8 @@ Method* ClassFileParser::parse_method(const ClassFileStream* const cfs,
   if (parsed_annotations.has_any_annotations())
     parsed_annotations.apply_to(methodHandle(THREAD, m));
 
-  if (is_hidden()) { // Mark methods in hidden classes as 'hidden'.
-    m->set_hidden(true);
+  if (is_hidden() && !m->is_hidden()) { // Mark methods in hidden classes as 'hidden'.
```

Comment:
This seems odd - how would m already be marked hidden? And why do we care? The check-and-branch is more expensive than just setting the field.

Reply:
This hidden flag is set by both the attribute during class file parsing and set to true if the whole class is hidden.  My original version of the assert_is_safe() function only allowed a flag to be set once, but I've changed it to only set to true.  So I can undo this.